### PR TITLE
Rbac- Test scram + bug fix on properties

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Updated one test scenario to test for sasl scram, discovered/ resolved two bugs
- sasl scram/plain set principal was using wrong dictionary key
- all components had redundant mds ssl props (removed them)
- ksql ssl props were incorrect

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
All rbac test scenarios pass
custom certs scenario, i added scram as the sasl mechanism bc it already wasnt doing mtls

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.control_center/molecule/rbac-scram-custom-1way/molecule.yml [hunk 1]
```diff
@@ -134,7 +134,8 @@ provisioner:
     group_vars:
       all:
         ssl_enabled: true
-        ssl_mutual_auth_enabled: true
+        # ssl_mutual_auth_enabled: true
```
Comment (domenicbove) [inline roles/confluent.control_center/molecule/rbac-scram-custom-1way/molecule.yml:137]: remove commented out code

File: roles/confluent.control_center/molecule/rbac-scram-custom-1way/molecule.yml [hunk 2]
```diff
@@ -134,7 +134,7 @@ provisioner:
     group_vars:
       all:
         ssl_enabled: true
-        ssl_mutual_auth_enabled: true
+        sasl_protocol: scram
 
         ssl_custom_certs: true
         ssl_ca_cert_filepath: roles/confluent.control_center/molecule/rbac-custom-certs-1way-mds/generated_ssl_files/ca.crt
```
Comment (domenicbove) [inline roles/confluent.control_center/molecule/rbac-scram-custom-1way/molecule.yml:140]: cert file path is wrong

**Comment by JumaX** [comment]
@domenicbove the test just completed successfully for me:

```    META: ran handlers
    META: ran handlers

    PLAY RECAP *********************************************************************
    localhost                  : ok=3    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0```

This was for rbac-scram-custom-1way.  What am I missing?

**Comment by domenicbove** [comment]
@JumaX its probably because the generated_ssl dirs gets gitiginored locally, but on jenkins with a fresh git clone there are no generated_ssl dirs and per my above comment the file paths were wrong in the molecule.yml vars

**Pull request merged**